### PR TITLE
Staking gate settings as a proof of concept

### PR DESCRIPTION
**Pull request by dvargasfuertes**
https://app.clarity.so/cabin/view/3039c279-2ee2-4da2-a604-dc1c23d5010c

Actual customization not yet implemented. This PR shows a proof of concept of staking and unstaking in the settings tab, where most of the customization would occur. A much broader product conversation/decision needs to be made about what gets gated and for how much

### CODE REVIEW

File: packages/app/components/constants.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { AbiType, StateMutabilityType, AbiItem } from "web3-utils";
 import {
+  // KOVAN_STAKING_ADDRESS,
   KOVAN_PASSPORT_FACTORY_ADDRESS,
+  // ROPSTEN_STAKING_ADDRESS,
   ROPSTEN_PASSPORT_FACTORY_ADDRESS,
 } from "@cabindao/nft-passport-contracts/artifacts/addresses";
```
Comment (dvargasfuertes) [inline packages/app/components/constants.ts:7]: I deployed the staking contract already to npm, kovan, ropsten just to get the front end development connected and not throwing errors, I expect the contracts themselves will have more iteration

File: packages/contracts/contracts/TestToken.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    event Fauceted(
+        uint256 quantity
+    );
+    function faucet(uint256 quantity) public
+    {
+        this.transferFrom(address(this), msg.sender, quantity* 10**uint(decimals()));
+        emit Fauceted(quantity); 
+    }
+
+    fallback() external {}
```
Comment (dvargasfuertes) [inline packages/contracts/contracts/TestToken.sol]: I'm having trouble getting the test token to show up in my metamask during testing

I keep seeing `Error: Transaction reverted: function selector was not recognized and there's no fallback function`. Any ideas?

File: packages/contracts/scripts/deploy.ts [hunk 3]
```diff
@@ -38,11 +40,12 @@ async function deploy(key: string) {
       `export declare const ${network.name.toUpperCase()}_${key}_ADDRESS: string;\n`
     );
   }
+  return contract.address;
 }
 
 async function main() {
   await deploy("PASSPORT_FACTORY");
-  await deploy("STAKING");
+  await deploy("TEST_TOKEN", "10000").then((addr) => deploy("STAKING", addr));
```
Comment (dvargasfuertes) [inline packages/contracts/scripts/deploy.ts]: when we eventually launch on main, we'll need to change this to point to the cabin token

File: packages/contracts/hardhat.config.ts [hunk 4]
```diff
@@ -21,6 +22,19 @@ task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
   }
 });
 
+task("faucet", "Seed an account with the test token")
```
Comment (dvargasfuertes) [inline packages/contracts/hardhat.config.ts:26]: helper to seed test token in development

**Comment by dvargasfuertes** [comment]
tagging @saitejareddysiddareddy for initial review

**Review by ssaitejar96 — APPROVED** [review]
LGTM!

**Pull request merged**